packages/react-core/src/components/Banner/Banner.tsx

```diff
@@ -1,39 +1,68 @@
 import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Banner/banner';
 import { css } from '@patternfly/react-styles';
+import { variantIcons } from '../Alert/AlertIcon';
+import { Flex, FlexItem } from '../../layouts';
 
 export interface BannerProps extends React.HTMLProps<HTMLDivElement> {
-  /** Content rendered inside the banner */
+  /** Content rendered inside the banner. */
   children?: React.ReactNode;
-  /** Additional classes added to the banner */
+  /** Additional classes added to the banner. */
   className?: string;
-  /** Variant styles for the banner */
-  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
+  /** A custom icon for the banner. This property will override the icon that is set based on
+   * the variant property.
+   */
+  customIcon?: React.ReactNode;
+  /** Flag for indicating whether the banner has a status icon. When set to "true", the icon
+   * will be set based on the variant property.
+   */
+  hasStatusIcon?: boolean;
   /** If set to true, the banner sticks to the top of its container */
   isSticky?: boolean;
-  /** Text announced by screen readers to indicate the type of banner. Defaults to "${variant} banner" if this prop is not passed in */
+  /** Text announced by screen readers to indicate the type of banner when the hasStatusIcon property
+   * is passed in. Defaults to "${variant} banner" if this property is not passed in.
+   */
   screenReaderText?: string;
+  /** Variant styles for the banner. */
+  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
 }
 
 export const Banner: React.FunctionComponent<BannerProps> = ({
   children,
   className,
+  customIcon,
+  hasStatusIcon = false,
   variant = 'default',
   screenReaderText,
   isSticky = false,
   ...props
-}: BannerProps) => (
-  <div
-    className={css(
-      styles.banner,
-      styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-      isSticky && styles.modifiers.sticky,
-      className
-    )}
-    {...props}
-  >
-    {children}
-    <span className="pf-u-screen-reader">{screenReaderText || `${variant} banner`}</span>
-  </div>
-);
+}: BannerProps) => {
+  const StatusIcon = variantIcons[variant];
+
+  return (
+    <div
+      className={css(
+        styles.banner,
+        styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
+        isSticky && styles.modifiers.sticky,
+        className
+      )}
+      {...props}
+    >
+      {hasStatusIcon ? (
+        <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+          <FlexItem>
+            <span className="pf-u-screen-reader">{screenReaderText || `${variant} banner`}</span>
```

Comment:
I think this is breaking since it works currently without `hasStatusIcon`.

Also is it just me or is the existing placement odd with the sr text coming after `children`? I think in other places, we put the sr text before the banner contents, assuming it provides context about the banner contents.

https://github.com/patternfly/patternfly-react/blob/44af73002bdeb1048be9b326abf7ffd4fcd474be/packages/react-core/src/components/Banner/Banner.tsx#L34-L37

Reply:
Re: the placement: yeah I noticed that as well and figured that was something worth updating in this PR.

Regarding the conditional rendering of the SR text: that *might* be iffy? With the way the SR text is currently used it's to convey the banner variant, though custom text can be passed into the `screenReaderText` prop to convey something other than status (like simply having SR text that says "Banner:").

Since we are moving away from using status variants to determine the banner color and such, it may come down to whether it would be considered fixing a bug (a banner that doesn't convey status shouldn't have hidden text that conveys status to SR's) or if it should be kept as-is.

Reply:
>If we're going to rely on color variants alone, and it's up to the dev to handle the status or whatever the banner is, it seems like we would want to always expose this prop, and/or show how to add screen reader text to the banner wherever it's appropriate to use. ie <Banner>banner text blah <span class="pf-screen-reader">sr text</span> apples and oranges</Banner>

If it would be breaking, we could possibly open an issue as part of the breaking change or v5 release. I could revert it so that the SR text always renders, but maybe just update the examples to either provide alternative, non-statusy sounding text ("This is a blue banner...") or making it so that you can have the SR text omitted (either passing in `null` or an empty string?). Wdyt?

>Just an aside, but do you think it makes sense to have screen reader text as a component of some sort? Currently that's part of the PF global CSS

That's an interesting idea... I might be on the fence about it? 😆 On the one hand, it could be relatively handy to just have a component you can pass in somewhere with its own props for things like full-width or absolute styling (or even as a variant of our existing Text component). 

On the other hand, I guess I'm not entirely sure whether its usage or benefit would be significant, and I might be wary about it being something exposed for public use without some good a11y docs. I did come across this, though: [Screen reader only component](https://paste.twilio.design/components/screen-reader-only/) so it's definitely something others have thought of.

Reply:
> If it would be breaking, we could possibly open an issue as part of the breaking change or v5 release. I could revert it so that the SR text always renders, but maybe just update the examples to either provide alternative, non-statusy sounding text ("This is a blue banner...") or making it so that you can have the SR text omitted (either passing in null or an empty string?). Wdyt?

Ah! I didn't notice that the sr text always renders - I thought it only rendered if you passed a value to the prop. Yeah, sounds good to me, especially since we are leaving the status variants in the component, and updating for v5.

It does beg the question of whether we should retain status variants alongside colors for uses like this, as opposed to replacing status with color. As in, have red/green/blue/etc and /danger/success/info, which would potentially allow for automatic injection of SR text (and other things we determine are useful) and potentially a separation of color global vars for blue/red/green and status global vars for info/danger/success.

Reply:
>It does beg the question of whether we should retain status variants alongside colors for uses like this, as opposed to replacing status with color.

I think the fact that anything could be passed into the Banner, including an icon, does make things a little more gray. The variants could still have some usefulness, but could arguably be replaced by a more fitting, existing prop (`screenReaderText` could just be passed in to render "Danger banner:" rather than needing to pass `variant="danger"`, but having that text auto-render could be useful and "variant" might be more easily understandable without a prop description)